### ISSUE TITLE
promtail can not parse cmd flag override configfile

### ISSUE THREAD

**Issue by daixiang0** [grafana/loki#638]
**Describe the bug**

**To Reproduce**
1. add config
```
# cat cmd/promtail/promtail-docker-config.yaml
server:
  http_listen_port: 9080
  grpc_listen_port: 0

positions:
  filename: /tmp/positions.yaml

clients:
  - url: http://loki:3100/api/prom/push
    batchwait: 3s <===
scrape_configs:
- job_name: system
  entry_parser: raw
  static_configs:
  - targets:
      - localhost
    labels:
      job: varlogs
      __path__: /var/log/*log
```
2. add value print in `pkg/promtail/client/config.go` and build
```
// UnmarshalYAML implement Yaml Unmarshaler
func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
    type raw Config
    var cfg raw 
    if c.URL.URL != nil {
        // we used flags to set that value, which already has sane default.
        cfg = raw(*c)
    } else {
        // force sane defaults.
        cfg = raw{
            BackoffConfig: util.BackoffConfig{
                MaxBackoff: 5 * time.Second,
                MaxRetries: 5,
                MinBackoff: 100 * time.Millisecond,
            },
            BatchSize: 100 * 1024,
            BatchWait: 1 * time.Second,
            Timeout:   10 * time.Second,
        }
    }   

    if err := unmarshal(&cfg); err != nil {
        return err 
    }   

    *c = Config(cfg)
    fmt.Println("config yaml", c.BatchWait)
    return nil 
}
```
3.set config in cmd line and run
```
./promtail --config.file=cmd/promtail/promtail-docker-config.yaml --client.batch-wait=1s
config yaml 3s
level=warn ts=2019-05-31T09:39:25.628290877Z caller=filetargetmanager.go:101 msg="WARNING!!! entry_parser config is deprecated, please change to pipeline_stages"
level=info ts=2019-05-31T09:39:25.629777023Z caller=server.go:120 http=[::]:9080 grpc=[::]:46530 msg="server listening on addresses"

```

**Expected behavior**
cmd line config should override configfile.

Additional context:
#520 
#636

**Comment by Kuqd** [comment]
This is the expected behaviour. You need to provide the --client.url and it will work.

**Comment by daixiang0** [comment]
Not sure, `--config`  should work with `--client.url` or another.

**Comment by daixiang0** [comment]
get it, add a comment to ensure users with no confuse about this.

**Comment by daixiang0** [comment]
@Kuqd any update?

**Comment by daixiang0** [comment]
@cyriltovena Hi, As #644 said, now can not use viper to do this.
How about use #639 for now?

**Comment by daixiang0** [comment]
any input? @cyriltovena

**Comment by daixiang0** [comment]
update

**Comment by sh0rez** [comment]
Resolved this in #980
